modules/age.nix

```diff
@@ -96,9 +109,27 @@ in {
       message = "age.sshKeyPaths must be set.";
     }];
 
+    # Create a new directory full of secrets for symlinking (this helps
+    # ensure removed secrets are actually removed, or at least become
+    # invalid symlinks).
+    system.activationScripts.agenixMountSecrets = ''
+      _count="$(basename "$(readlink /run/secrets)" || echo 0)"
+      (( ++_count ))
+      echo "[agenix] symlinking new secrets generation $_count to /run/secrets..."
+      mkdir -pm 0750 "${cfg.secretsMountPoint}"
+      mount | grep "${cfg.secretsMountPoint} type ramfs" -q || mount -t ramfs none "${cfg.secretsMountPoint}" -o nodev,nosuid,mode=0750
+      mkdir -pm 0750 "${cfg.secretsMountPoint}/$_count"
+      chown :keys "${cfg.secretsMountPoint}" "${cfg.secretsMountPoint}/$_count"
+      (( _count > 1 )) && rm -rf "$(readlink /run/secrets)"
```

Comment:
Maybe it would be better to `rm -rf "${cfg.secretsMountPoint/(( _count - 1 ))"` instead, in case some malicious user / program / whatever symlinks /run/secrets to someplace you probably don't want to nuke...

Or, at the cost of even more complexity, remove all secrets individually, and then `rmdir "${cfg.secretsMountPoint/(( _count - 1 ))"`, to ensure it's actually empty.

Any thoughts?